core/src/main/java/tech/tablesaw/api/NumberColumn.java

```diff
@@ -299,22 +747,339 @@ default double kendalls(NumberColumn otherColumn) {
      * Returns the number of unique values in this column, excluding missing values
      */
     @Override
-    default int countUnique() {
+    public int countUnique() {
         DoubleSet doubles = new DoubleOpenHashSet();
         for (int i = 0; i < size(); i++) {
-            if (!NumberColumn.valueIsMissing(getDouble(i))) {
+            if (!data.isMissingValue(getDouble(i))) {
                 doubles.add(getDouble(i));
             }
         }
         return doubles.size();
     }
 
-    default Selection isMissing() {
+    public Selection isMissing() {
         return eval(isMissing);
     }
 
-    default Selection isNotMissing() {
+    public Selection isNotMissing() {
         return eval(isNotMissing);
     }
 
+    @Override
+    public NumberColumn appendObj(Object obj) {
+        if (!(obj instanceof Double)) {
+            throw new IllegalArgumentException();
+        }
+        return append((double) obj);
+    }
+
+    /**
+     * Counts the number of rows satisfying predicate, but only upto the max value
+     *
+     * @param test the predicate
+     * @param max  the maximum number of rows to count
+     * @return the number of rows satisfying the predicate
+     */
+    public int count(DoublePredicate test, int max) {
+        int count = 0;
+        for (int i = 0; i < size(); i++) {
+            if (test.test(getDouble(i))) {
+                count++;
+                if (count >= max) {
+                    return count;
+                }
+            }
+        }
+        return count;
+    }
+
+    /**
+     * Returns a new NumberColumn with only those rows satisfying the predicate
+     *
+     * @param test the predicate
+     * @return a new NumberColumn with only those rows satisfying the predicate
+     */
+    public NumberColumn filter(DoublePredicate test) {
```

Comment:
I wonder if this is an operation we can implement on the data wrappers like `lead` and `lag`